### PR TITLE
DESI

### PR DESCRIPTION
**Pull request by dkirkby**
Work in progress on implementing the baseline DESI optics model.  Still needs tilts for ADC surfaces.

### CODE REVIEW

**Comment by dkirkby** [comment]
@jmeyers314 I think this is ready for review.

**Comment by dkirkby** [comment]
I am updating the unit tests to add coverage of the new coordSys rotX,Y,Z parsing...

I had to comment out `addopts = -n auto`  in  `setup.cfg` since my version of `pytest` does not support this option.

There seems to be at least one slow notebook in `test_notebooks.py` that I did not track down.

There are two deprecations warnings that I did not fix:
```
YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
```
and
```
DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
```

File: batoid/parse.py [hunk 1]
```diff
@@ -39,9 +42,17 @@ def parse_coordSys(config, coordSys=batoid.CoordSys()):
         shift = [x, y, z]
     elif 'shift' in config:
         shift = config.pop('shift')
-    # Leaving rotation out for the moment...
     if shift != [0.0, 0.0, 0.0]:
         coordSys = coordSys.shiftLocal(shift)
+    # At most one (nonzero) rotation can be included and is applied after the shift.
+    rotXYZ = np.deg2rad([config.pop('rot' + axis, 0.0) for axis in 'XYZ'])
```
Comment (jmeyers314) [inline batoid/parse.py]: For the moment I think I'd prefer to use radians directly in the yaml to be consistent with, e.g., the arguments to `ObscRectangle`.

File: tests/test_optic.py [hunk 2]
```diff
@@ -114,6 +115,22 @@ def test_shift():
         assert telescope == shifted
 
 
+@timer
+def test_rotXYZ_parsing():
```
Comment (jmeyers314) [inline tests/test_optic.py]: Need to add this to the `if __name__ == '__main__':` block.

File: batoid/parse.py (hunk 1 above)
Comment (dkirkby) [reply batoid/parse.py]: Ok, done.

File: tests/test_optic.py (hunk 2 above)
Comment (dkirkby) [reply tests/test_optic.py]: Done.

**Comment by dkirkby** [comment]
The unit tests ran earlier today, so maybe the latest failure of all tests was just a glitch.

My local pytest is running fine (after removing `-n auto`)

**Comment by jmeyers314** [comment]
My local pytest is running fine now too, and I see at least the python3.6 travis branches are passing.  That's good enough for me; merging now and I'll fight with travis later if it's still a problem.

**Pull request merged**